arch/xtensa/Kconfig

```diff
@@ -93,6 +93,24 @@ config XTENSA_NUM_SPIN_RELAX_NOPS
 	  Specify the number of NOPs in Xtensa specific
 	  arch_spin_relax().
 
+config CPU_HAS_DSP
+	bool
+	help
+	  This option is enabled when the CPU has a hardware DSP unit such
+	  as a HiFi4.
+
+if CPU_HAS_DSP
+menu "DSP Options"
+
+config DSP_SHARING
+	bool "DSP register sharing"
+	help
+	  This option enables preservation of the hardware DSP registers
+	  across context switches to allow multiple threads to perform
+	  concurrent DSP operations.
```

Comment:
So, I get why this is called "SHARING" by symmetry with the existing FPU features.  But with Xtensa, existing toolchains will generate HiFi instructions that use the AE registers whether or not the app wants them to[1].  There's really no feasible situation where we'd want this disabled (or rather, it would be feasible, but would require apps directly control compiler flags and probably enforce their own ABIs).

Suggest removing the kconfig here and just keying off of XCHAL_HAVE_xxx from core-isa.h

[1] gcc doesn't, and in general xt-clang won't for scalar code, but if you look at the generated assembly for anything that looks like array processing you'll see HiFi bundles in the output.

Reply:
I still like the "SHARING" for purposes of symmetry. However, you raise a very interesting point with regards to toolchain behaviour. I had originally set "SHARING" to 'n' so that we would not suffer stack size nor context switch runtime penalties. However, it might be worthwhile to have the default set according to the board or soc.

Reply:
I've been doing some more reading on this, and it looks like xt-clang may automatically generate HiFi code if any of the following are met ...

1. Automatic vectorization is enabled (via `O3, -LNO:simd`, or `-fvectorize`, or )
2.  Or if the `-mcoproc` option is used.

Near as I can tell, these are not enabled by default in Zephyr. Furthermore, I do not yet see any options for them to be enabled in Zephyr. Assuming the both the documentation and my interpretation are correct, then it may be reasonable _at this point in time_ to have the project be responsible for the _sharing_ and compilation flags.